types/meow/index.d.ts

```diff
@@ -33,14 +35,32 @@ declare namespace meow {
         booleanDefault?: boolean | null;
     }
 
-    interface Result {
+    interface Result<O extends Options, F = $PropertyType<O, 'flags'>> {
         input: string[];
-        flags: { [name: string]: any };
+        flags: {
+            [id in keyof F]: F[id] extends GenericOption<infer T>
```

Comment:
This is not good enough since in the result flags are also converted to camelCase (https://github.com/sindresorhus/meow#features)

Reply:
with this it should be good right ? https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29657/files#diff-7f2fb2019c1ea4e923665a9a05b41321R44

Reply:
But the typings are not retained for camel case conversion.
`cli4.flags.asd // error` is also not going to error. 
So I don't see any reason why this PR makes sense.

Reply:
well, at least you have property inference with this PR, but ok if it doesn't make sense you can close it